### DISCUSSION TITLE
Default to spotify library

### DISCUSSION THREAD

**Discussion by RayLation** [music-assistant/support#1550] [Q&A] [answered]
I have a few artists that are in my favorites in spotify. But I didn't favorite any album/tracks for them.
When I click on the artist in mass, no tracks/albums show up in the list.
But then, if I click on the "library button" (see screenshot), it goes to a different view with what looks like full content from spotify.
Is there a way to get to this view by default?

![image](https://github.com/music-assistant/hass-music-assistant/assets/92007174/81327e71-7e14-45a0-b9da-5f267119aa36)

![image](https://github.com/music-assistant/hass-music-assistant/assets/92007174/3d55e936-f47a-42be-9cf4-8cb54e5e5fc9)

**Comment by OzGav** [comment]
Not at the moment. The current concept is MA shows you what you have in your library. That library is everything that you have on a local server plus everything you have in the music provider libraries. As you have found you can see everything in the music provider database by either toggling that filter or you can also do a global search. From there you can add to your library

**Comment by arretx** [comment]
I'm with @RayLation.  I find myself needing to re-select this every time I want to play something.  I don't have any local files other than mp3 sound effects that are designed for specific automations.

**Reply by OzGav** [reply]
You write “reselect this” but I don’t know what “this” is? You have to add items to the library to see them in MA, that is how it works. If you don’t want to add to the library then go to the global search and you can just search your provider.

**Reply by RayLation** [reply]
in the dropdown menu (in the screenshot), you can select "Library" or "Spotify".
We want it to default to "Spotify" because this is where the music is.
Right now, by default it's always "Library", so we need to manually select "Spotify" everytime.

**Reply by OzGav** [reply]
Ok understand the request. This is not how MA works. It is designed to show you what you have in your library. That is, the music that you like. 

Why is the global search view not doing what you want?
![IMG_1136](https://github.com/music-assistant/hass-music-assistant/assets/19848947/803b4164-0e9b-4c0f-a706-031a8101617f)

**Reply by RayLation** [reply]
If I search for an artist in global search and then click on an artist to go to the artist's page, it is sometimes empty, like in the example below for Avenged Sevenfold. I don't understand why. In those cases, there's a menu that appears to lets me choose between "my library" or "spotify" . When I select spotify, the tracks/albums magically appear. I'd like it to default to spotify for those cases. 

For other bands, like example for Metallica, the artist's page shows all tracks/albums in spotify by default and the menu to switch from "my library" and "spotify" is not even there (a good thing!)

Avenged:
![image](https://github.com/music-assistant/hass-music-assistant/assets/92007174/d5582360-5473-4a30-91f5-c71856e4a20c)
![image](https://github.com/music-assistant/hass-music-assistant/assets/92007174/2316a47d-bdba-44d3-ab97-527bc0dc8d08)
![image](https://github.com/music-assistant/hass-music-assistant/assets/92007174/cffd093d-bfa5-4ad0-b7e5-d7a038b98a58)

**Comment by RayLation** [comment]
so it seems this happens for artists that I added to my library but for which I didn't add songs/albums to library. So should I go ahead and remove everything from my library so that I can see all tracks/albums from any artist by default? (Since I don't have time to manually add all tracks/songs one by one)

**Reply by OzGav** [reply, marked as answer]
Correct. That is the normal behaviour and exactly what happens in Spotify if you look at your library there. There is a problem to be resolved when adding an album that all tracks aren’t automatically added, this will be fixed soon.

So yes if you remove the Artist from the Spotify library (and thus the MA library) then MA will show you what is available on the provider (as the library is empty)

**Reply by RayLation** [reply]
OK...
Is there a way to that adding an artist adds all albums/tracks for this artist (Like adding an album should add all tracks in this album)?

or at least a way to mass-select albums/tracks from an artist?

**Reply by OzGav** [reply]
Not at the moment. Previous requests were for the ability to play all tracks from an artist if you "play Artist" so that is coming. What you are asking for is not the intent of MA. Whilst you are using just one provider some people have many and adding all music from all providers is not desirable behaviour. 

As for mutiple add you can already do this. Click on the "Select Multiple Items" icon at the top of the album list, select the albums you want and then select ACTIONS in the popup at the bottom.

**Reply by RayLation** [reply]
thanks for you time sir !

I guess Music Assistant appeals to me for the wrong reasons. I saw a great Spotify controller (better then the built-in one in hass), but it is not meant exactly for this.

**Reply by OzGav** [reply]
Keep your eye on the project. The UI is constantly improving and bugs fixed. If this request gets enough votes from other users then it may get looked at further.
